Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -48,6 +49,15 @@ public override void Initialize()
             SubscribeLocalEvent<SalvageMagnetComponent, InteractHandEvent>(OnInteractHand);
             SubscribeLocalEvent<SalvageMagnetComponent, ExaminedEvent>(OnExamined);
             SubscribeLocalEvent<SalvageMagnetComponent, ComponentShutdown>(OnMagnetRemoval);
+            SubscribeLocalEvent<GameRunLevelChangedEvent>(OnRoundEnd);
+        }
+
+        private void OnRoundEnd(GameRunLevelChangedEvent ev)
+        {
+            if(ev.New == GameRunLevel.PostRound)
+            {
+                _salvageGridStates.Clear();
+            }
```

Comment:
Use `RoundRestartCleanupEvent` instead, it's specifically made for round-restart cleanup.